tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs

```diff
@@ -149,6 +205,77 @@ void UpsertVstsBugToDb(SqlConnection sqlConnection, BugQuery bugQuery, int bugCo
                 CommandText = "UpsertVstsBug"
             };
 
+            cmd.Parameters.Add(new SqlParameter("@BuildId", buildId));
+            cmd.Parameters.Add(new SqlParameter("@BugId", bugId));
+
+            cmd.ExecuteNonQuery();
+        }
+        IList<VstsBuild> FilterBuildsByExistingBugs(IList<VstsBuild> builds)
+        {
+            IList<VstsBuild> filteredBuilds = new List<VstsBuild>();
+            SqlConnection sqlConnection = null;
+            try
+            {
+                sqlConnection = new SqlConnection(this.dbConnectionString);
+                sqlConnection.Open();
+
+                foreach (VstsBuild build in builds)
+                {
+                    var cmd = new SqlCommand
+                    {
+                        Connection = sqlConnection,
+                        CommandType = CommandType.StoredProcedure,
+                        CommandText = "QueryVstsBugsForMatchingBuild"
+                    };
+
+                    cmd.Parameters.Add(new SqlParameter("@BuildId", build.BuildId));
+
+                    SqlDataReader reader = cmd.ExecuteReader();
+                    if (!reader.HasRows)
+                    {
+                        filteredBuilds.Add(build);
+                    }
+
+                    reader.Close();
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            finally
+            {
+                sqlConnection?.Close();
+            }
+
+            return filteredBuilds;
+        }
+
+        // We don't want to spam making bugs.
+        // Given this logic runs at least every few minutes, only make a bug for builds in the past 1 hour.
```

Comment:
What causes this logic to run every few minutes? Is it scheduled? Or is it triggered each time a build completes?

Reply:
This loop:
https://github.com/and-rewsmith/iotedge/blob/andsmi/vsts-sync-bug-automation/tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs#L41